language/module.md

```diff
@@ -45,6 +45,12 @@ already explains what a module is and what it can contain.
 | `/lib/puppet/type/`   | Custom types |
 | `/lib/puppet/provider/` | Custom provider for one or multiple types |
 | `/lib/puppet/functions/` | Modern functions in Ruby for the new API |
+| `/lib/puppet/datatypes/` | Custom Puppet Datatypes |
```

Comment:
is that the same as `/lib/puppet/type/`?

Reply:
No, `/lib/puppet/type/` is for Resource Types. `/lib/puppet/datatypes/` is for custom data types.  "Type" is overused in Puppet, but ah well.  You can see an example here: https://github.com/hlindberg/tahu/tree/master/lib/puppet/datatypes/tahu

Reply:
ahhh those types! :D

Reply:
I pushed an update which clarifies "Resource types" in the type and provider lines.